synapse/metrics/background_process_metrics.py

```diff
@@ -92,7 +97,11 @@ def collect(self):
             labels=["name"],
         )
 
-        for desc, processes in six.iteritems(_background_processes):
+        # We copy the dict so that it doesn't change from underneath us
+        with _bg_metrics_lock:
+            _background_processes_copy = dict(_background_processes)
```

Comment:
I feel like `dict(_background_processes)` shouldn't require a lock?

Reply:
I'm completely failing to find any documentation about what is and isn't safe TBH